src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -208,12 +208,13 @@ public class ConsensusManager : IConsensusManager, IDisposable
             //  coinview and it will abstract the methods `RewindAsync()` `GetBlockHashAsync()`
 
             uint256 consensusTipHash = await this.consensusRules.GetBlockHashAsync().ConfigureAwait(false);
+            bool blockStoreDisabled = this.chainState.BlockStoreTip == null;
 
             while (true)
             {
                 this.Tip = chainTip.FindAncestorOrSelf(consensusTipHash);
 
-                if (this.Tip?.HashBlock == consensusTipHash)
+                if ((this.Tip != null) && (blockStoreDisabled || (this.chainState.BlockStoreTip.Height >= this.Tip.Height)))
```

Comment:
Not sure about this at all, but it is probably a subject to change soon

Reply:
I remember we had some big discussions here.
This reverts back to how it is on master right now, so I will leave it as is and we can revisit when it will get changed.